Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -608,15 +531,22 @@ namespace AzToolsFramework
         for (size_t arrayIndex = 0, numIndices = domArray.ArraySize(); arrayIndex < numIndices; ++arrayIndex)
         {
             auto& childValue = domArray[arrayIndex];
-            AddChildFromDomValue(childValue, aznumeric_cast<int>(arrayIndex));
+            AddChildFromDomValue(childValue, arrayIndex);
         }
     }
 
     void DPERowWidget::HandleOperationAtPath(const AZ::Dom::PatchOperation& domOperation, size_t pathIndex)
     {
         const auto& fullPath = domOperation.GetDestinationPath();
         auto pathEntry = fullPath[pathIndex];
-        AZ_Assert(pathEntry.IsIndex() || pathEntry.IsEndOfArray(), "the direct children of a row must be referenced by index");
+
+        const bool isValidIndex = pathEntry.IsIndex() || pathEntry.IsEndOfArray();
```

Comment:
Could you clarify when this is used? It's fully valid to patch a property of a row (e.g. to update the `ForceAutoExpand` property), but it looks like this is maybe only called with the root of a patch so you're just saying the very first patch entry must be an index because we don't have any adapter node attributes that would end up here which... I think is true, but it might be possible to make this clearer.

Reply:
yikes. No, this is a real bug, albeit one we've always had. I'll add some code to address this.

Reply:
fixed, now clearing row and setting it from the corresponding DOM node. Might be overkill, but since this happens seldom to never, I think safety is the right approach.